pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -703,7 +703,14 @@ public synchronized PinotResourceManagerResponse deleteSegments(String tableName
       Preconditions.checkArgument(TableNameBuilder.isTableResource(tableNameWithType),
           "Table name: %s is not a valid table name with type suffix", tableNameWithType);
       HelixHelper.removeSegmentsFromIdealState(_helixZkManager, tableNameWithType, segmentNames);
-      _segmentDeletionManager.deleteSegments(tableNameWithType, segmentNames);
+      TableConfig tableConfig = ZKMetadataProvider.getTableConfig(_propertyStore, tableNameWithType);
+      String deletedSegmentRetentionString = tableConfig.getValidationConfig().getDeletedSegmentRetentionInDays();
+      if (deletedSegmentRetentionString != null) {
+        _segmentDeletionManager.deleteSegments(tableNameWithType, segmentNames,
+            Integer.parseInt(deletedSegmentRetentionString));
```

Comment:
Add a try catch around this value parsing. Log a warning if the value is invalid, then fall back to the default retention. We don't want to fail the segment deletion by an invalid retention config

Reply:
+1